src/dune_lang/package_variable_name.mli

```diff
@@ -32,6 +32,10 @@ val build : t
 val dev : t
 val one_of : t -> t list -> bool
 
+(** The set of variable names whose values are expected to differ depending on
+    the current platform. *)
+val platform_specific : Set.t
```

Comment:
Looking at this again, it seems awkward that we can't generate multiple configurations for variables outside of this set.

Reply:
I'll take a closer look at this. I don't know off hand what the consequence would be of allowing arbitrary solver variables to be configured.